test/profiling/integration.test.ts

```diff
@@ -351,6 +351,26 @@ describe('profiling integration', () => {
       jest.runAllTimers();
     });
   });
+
+  test('platformProviders flag passed down to native', () => {
+    mock = initTestClient({ withProfiling: false });
```

Comment:
I've noticed that this test is failing. My understanding is that this is only a test issue and does not have impact on the functional code. I experimented with changing the `withProfiling` parameter to `true` and added a new parameter in the `initTestClient` to avoid using the default value for the `HermesProfilingOptions`.
I'm attaching a patch with the above.

[integration.test.patch](https://github.com/user-attachments/files/17024543/integration.test.patch)

Feel free to ignore if this is in the wrong direction 😅

Reply:
hi, and welcome to Sentry 🙂 interestingly the test works when running it in isolation, so I assume there's some shared state that breaks it when running the whole suite. I applied your patch in https://github.com/getsentry/sentry-react-native/pull/4094/commits/841d5b70f133984afd61fb71188c5aeeaa7ab32b, thanks for this!